### PR TITLE
[web] Integration test for selectable text

### PR DESCRIPTION
**Pull request by mdebbar**
## Description

Integration test to make sure selectable text works end-to-end with the web engine.

### CODE REVIEW

File: e2etests/web/regular_integration_tests/test_driver/text_editing_integration.dart [hunk 1]
```diff
@@ -141,6 +142,73 @@ void main() {
...
+    final Offset topLeft = selectable.localToGlobal(Offset.zero);
+    final Offset topRight =
+        selectable.localToGlobal(Offset(selectable.size.width, 0.0));
+
+    // Drag by mouse to select the entire selectable text.
+    TestGesture gesture =
+        await tester.startGesture(topLeft, kind: PointerDeviceKind.mouse);
+    addTearDown(gesture.removePointer);
+    await gesture.moveTo(topRight);
+    await gesture.up();
```
Comment (nturgut) [inline e2etests/web/regular_integration_tests/test_driver/text_editing_integration.dart:166]: Wuhuuu :D

**Comment by nturgut** [comment]
After the https://github.com/flutter/flutter/pull/65499 and the engine change is merged. We can add one more method to test the changing input configuration. (We need to sync to head to be able to fetch the framework change though, since the younger framework commit older than the engine would be fetched).

Up to you though, it can also be another PR :)

**Comment by nturgut** [comment]
> After the [flutter/flutter#65499](https://github.com/flutter/flutter/pull/65499) and the engine change is merged. We can add one more method to test the changing input configuration. (We need to sync to head to be able to fetch the framework change though, since the younger framework commit older than the engine would be fetched).
> 
> Up to you though, it can also be another PR :)

Looks like it is failing for firefox, if we run using debug we will have a more info on logs. 

```
flutter drive --target=test_driver/text_editing_integration.dart -d web-server --browser-name=firefox --headless --local-engine=host_debug_unopt
```

Have you checked the failure locally, is it because of the unmerged PRs or is it a firefox specific issue?

**Comment by mdebbar** [comment]
@nturgut it turns out `window.getSelection()` in firefox doesn't include selection inside `<input>` and `<textarea>` elements.

https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection
> It is worth noting that currently getSelection() doesn't work on the content of <textarea> and <input> elements in Firefox, Edge (Legacy) and Internet Explorer. HTMLInputElement.setSelectionRange() or the selectionStart and selectionEnd properties could be used to work around this.

The good news is, this only affects the test, the app actually works fine on firefox. I'll change the test so we read the selection from the DOM in a way that works in all browsers.

**Pull request merged**